File: library/src/main/java/com/mikepenz/storyblok/model/Datasource.java
package com.mikepenz.storyblok.model;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by mikepenz on 14/04/2017.
 */

public class Datasource extends Entity {
    private String value;

    public Datasource(JSONObject story) {
        super(story);

        if (story.has("value")) {
            this.value = story.optString("value");
        }
    }

    public static List<Datasource> parseDatasource(JSONObject result) {
        if (result != null && result.has("datasource_entries")) {
            JSONArray jsonArray = result.optJSONArray("datasource_entries");
            List<Datasource> datasources = new ArrayList<>(jsonArray.length());
            for (int i = 0; i < jsonArray.length(); i++) {
                datasources.add(new Datasource(jsonArray.optJSONObject(i)));
            }
            return datasources;
        }
        return null;
    }

    public String getValue() {
        return value;
    }
}

File: library/src/main/java/com/mikepenz/storyblok/model/Datasource.kt
package com.mikepenz.storyblok.model

import org.json.JSONObject
import java.util.*

/**
 * Created by mikepenz on 14/04/2017.
 */

class Datasource(story: JSONObject) : Entity(story) {
    var value: String? = null
        private set

    init {

        if (story.has("value")) {
            this.value = story.optString("value")
        }
    }

    companion object {

        fun parseDatasource(result: JSONObject?): List<Datasource>? {
            if (result != null && result.has("datasource_entries")) {
                val jsonArray = result.optJSONArray("datasource_entries")
                val datasources = ArrayList<Datasource>(jsonArray.length())
                for (i in 0 until jsonArray.length()) {
                    datasources.add(Datasource(jsonArray.optJSONObject(i)))
                }
                return datasources
            }
            return null
        }
    }
}
